api/envoy/api/v3alpha/core/config_source.proto

```diff
@@ -161,4 +172,7 @@ message ConfigSource {
   // means no timeout - Envoy will wait indefinitely for the first xDS config (unless another
   // timeout applies). The default is 15s.
   google.protobuf.Duration initial_fetch_timeout = 4;
+
+  // API version for xDS endpoint
+  XdsApiVersion version = 6;
```

Comment:
Nit: call this field explicitly `xds_api_version` to distinguish from other version concerns.